solidity/core/test/validator-manager/multisigValidatorManager.test.ts

```diff
@@ -0,0 +1,264 @@
+      validatorSigner1,
+      validatorSigner2,
+      validatorSigner3,
+    ] = signers;
+    validator0 = await Validator.fromSigner(validatorSigner0, OUTBOX_DOMAIN);
+    validator1 = await Validator.fromSigner(validatorSigner1, OUTBOX_DOMAIN);
+    validator2 = await Validator.fromSigner(validatorSigner2, OUTBOX_DOMAIN);
+    validator3 = await Validator.fromSigner(validatorSigner3, OUTBOX_DOMAIN);
+  });
+
+  beforeEach(async () => {
+    const validatorManagerFactory = new TestMultisigValidatorManager__factory(
+      signer,
+    );
+    validatorManager = await validatorManagerFactory.deploy(
+      OUTBOX_DOMAIN,
+      [validator0.address],
+      QUORUM_THRESHOLD,
+    );
+  });
+
+  describe('#constructor', () => {
+    it('sets the outboxDomain', async () => {
+      expect(await validatorManager.outboxDomain()).to.equal(OUTBOX_DOMAIN);
+    });
+
+    it('sets the outboxDomainHash', async () => {
+      const outboxDomainHash = await validatorManager.domainHash(OUTBOX_DOMAIN);
+      expect(await validatorManager.outboxDomainHash()).to.equal(
+        outboxDomainHash,
+      );
+    });
+
+    it('enrolls the validator set', async () => {
+      expect(await validatorManager.validatorSet()).to.deep.equal([
+        validator0.address,
+      ]);
+    });
+
+    it('sets the quorum threshold', async () => {
+      expect(await validatorManager.quorumThreshold()).to.equal([
+        QUORUM_THRESHOLD,
+      ]);
+    });
+  });
+
+  describe('#enrollValidator', () => {
+    it('enrolls a validator into the validator set', async () => {
+      await validatorManager.enrollValidator(validator1.address);
+
+      expect(await validatorManager.validatorSet()).to.deep.equal([
+        validator0.address,
+        validator1.address,
+      ]);
+    });
+
+    it('emits the EnrollValidator event', async () => {
+      expect(await validatorManager.enrollValidator(validator1.address))
+        .to.emit(validatorManager, 'EnrollValidator')
+        .withArgs(validator1.address);
+    });
+
+    it('reverts if the validator is already enrolled', async () => {
+      await expect(
+        validatorManager.enrollValidator(validator0.address),
+      ).to.be.revertedWith('already enrolled');
+    });
+
+    it('reverts when called by a non-owner', async () => {
+      await expect(
+        validatorManager.connect(nonOwner).enrollValidator(validator1.address),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('#unenrollValidator', () => {
+    beforeEach(async () => {
+      // Enroll a second validator
+      await validatorManager.enrollValidator(validator1.address);
+    });
+
+    it('unenrolls a validator from the validator set', async () => {
+      await validatorManager.unenrollValidator(validator1.address);
+
+      expect(await validatorManager.validatorSet()).to.deep.equal([
+        validator0.address,
+      ]);
+    });
+
+    it('emits the UnenrollValidator event', async () => {
```

Comment:
Consider renaming the events `ValidatorEnrolled`, `ValidatorUnenrolled`. We should modify the event name in `Router` to be conssitent, as it differs from the pattern in `XAppConnectionManager`

Reply:
It looks like `Router` isn't the only contract with event names that aren't in the past tense, it's kinda split but I think past tense events are the minority. Maybe we should all agree on a pattern and do the rename in a separate tiny PR?